File: Ochinpo.java
import java.util.Random;

public class Ochinpo
{
    public static void main (String[] args)
    {
	String[] str = {"お", "ち", "ん", "ぽ"};
	int rnd = 0;
	int cnt = 0;
	int complete = 0;
	Random makeRnd = new Random();
	
	while (complete < 4) {
	    cnt++;
	    rnd = makeRnd.nextInt(4);
	    System.out.print(str[rnd]);
	    complete = (rnd == complete) ? ++complete : 0;
	}
	
	System.out.print("\n");
	System.out.println(cnt + "回目で完成しました");
	System.exit(0);
    }
}

File: Objective-C/Ochinpo.m
#import <Foundation/Foundation.h>

@interface Ochinpo : NSObject

@end

@implementation Ochinpo : NSObject
- (void) ochinchin {
    srand(time(nil));
    int rnd = 0;
    int complete = 0;
    int cnt = 0;
    NSString *ocnp[] = {@"お", @"ち", @"ん", @"ぽ"};

    while (complete < 4) {
        cnt++;
        rnd = rand()%4;
        printf("%s", [ocnp[rnd] UTF8String]);
        complete = (complete == rnd) ? ++complete : 0;
    }

    printf("\n");
    printf("%d回目で完成しました\n", cnt);
}
@end

